documentation/release_notes.markdown

```diff
@@ -7,9 +7,126 @@ canonical: "/puppetserver/latest/release_notes.html"
+
+```
+java.lang.IllegalStateException: Cannot initialize master with partial state; need all files or none.
+Found:
+/var/lib/puppet/ssl/private_keys/master.pem
+Missing:
+/var/lib/puppet/ssl/certs/master.pem
+```
+
+In this situation, Puppet Server 2.7 now uses pre-generated public and private keys to generate a certificate for the master and will finish starting without error.
+
+-   [SERVER-528](https://tickets.puppetlabs.com/browse/SERVER-528)
+
+### New Feature: Required gems are packaged with Puppet Server, with a new `GEM_PATH` and setting
+
+Some gems are required by both the Puppet agent and Puppet Server. To ship these gems as part of our packaging, Puppet Server 2.7 changes how Puppet Server looks for gems.
+
+In Puppet Server 2.6 and earlier, Puppet Server's `GEM_PATH` was comprised of only a single directory by default: `/opt/puppetlabs/server/data/puppetserver/jruby-gems`. Puppet Server also used this directory as the value for `GEM_HOME`, meaning that the `puppetserver gem install` command installed gems to this directory.
+
+In Puppet Server 2.7, we've added a second path to `GEM_PATH`: `/opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems`. Gems that are known to be needed by Puppet Server will be installed in this directory as part of the Puppet Server packaging.
+
+`GEM_HOME` still points to the same `jruby-gems` directory as it did in previous releases, and `puppetserver gem install` continues to install gems to, and use gems from, that directory.
+
+To configure the `GEM_PATH`, set the new `gem-path` setting in [`puppetserver.conf`][puppetserver.conf].
+
+-   [SERVER-1412](https://tickets.puppetlabs.com/browse/SERVER-1412)
+
+### Bug Fix: `puppetserver gem` command makes installed gems readable
+
+In Puppet Server 2.6 and earlier, if the system's default umask did not permit world-readability for gems installed with the 'puppetserver gem' subcommand, the `puppetserver` process might not be able to use the resulting gemspec files, leading to errors such as:
+
+```
+Exception in thread "main" org.jruby.exceptions.RaiseException: (LoadError) no such file to load -- trollop
+Exception in thread "main" org.jruby.exceptions.RaiseException: (Errno::EACCES) /opt/puppetlabs/server/data/puppetserver/jruby-gems/specifications/trollop-2.1.2.gemspec
+```
+
+Puppet Server 2.7 resolves this issue by explicitly setting a umask of 0022 when running any `puppetserver gem` subcommand. This ensures that `puppetserver` can use any gems installed by the `gem` subcommand at run-time.
+
+-   [SERVER-1601](https://tickets.puppetlabs.com/browse/SERVER-1601)
+
+### Other new features
+
+-   [SERVER-1585](https://tickets.puppetlabs.com/browse/SERVER-1585): Set JRuby `compat-version` in [`puppetserver.conf`][puppetserver.conf].
```

Comment:
Might be worth wordsmithing this a little more.  This adds the ability for users to configure Puppet Server to experimentally run under an MRI 2.0 compatibility mode, rather than just the default 1.9.3 mode.  I think it's worth calling out the "2.0" version in this description somewhere.  Since several users have been asking about it, I'd also probably move this up toward the top of the list of new stuff for better visibility